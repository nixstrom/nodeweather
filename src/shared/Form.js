import React from 'react'

const Form = ({ onSubmit, cityInputRef }) => (
  <form className="form-inline" onSubmit={ onSubmit}>
    <div className="form-group">
      <input type="text" className="form-control" id="city" placeholder="City" ref={cityInputRef} />
    </div>
    <button type="submit" className="btn btn-default">Search</button>
  </form>)


export default Form
