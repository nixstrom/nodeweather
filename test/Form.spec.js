import React from 'react'
import { shallow } from 'enzyme'
//import sinon from 'sinon'

import Form from '../src/shared/Form'

describe('<Form />', () => {
  const mockSubmitHandler = jest.fn()

  let wrapper
  beforeEach(() => {
    wrapper = shallow(<Form
        onSubmit={mockSubmitHandler}
        cityInputRef={() => {}}
      />)
  })
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('triggers submit handler', () => {
    wrapper.simulate('submit')
    expect(mockSubmitHandler).toHaveBeenCalled()
  })
})
