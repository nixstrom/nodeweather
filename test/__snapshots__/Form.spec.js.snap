// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders correctly 1`] = `
<form
  className="form-inline"
  onSubmit={[MockFunction]}
>
  <div
    className="form-group"
  >
    <input
      className="form-control"
      id="city"
      placeholder="City"
      type="text"
    />
  </div>
  <button
    className="btn btn-default"
    type="submit"
  >
    Search
  </button>
</form>
`;
